hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -174,6 +179,21 @@ public void handleClientMessage(ClientMessage clientMessage, Connection connecti
         }
     }
 
+    private MessageTask resolveClientMessage(ClientMessage clientMessage, Connection connection) {
```

Comment:
I'm not crazy about it; especially when factory count increases. But also the litter; since you get an additional iterator instance.